rest-api-spec/src/main/resources/rest-api-spec/api/put_script.json

```diff
@@ -11,10 +11,9 @@
           "description" : "Script ID",
           "required" : true
         },
-        "lang" : {
+        "context" : {
           "type" : "string",
-          "description" : "Script language",
-          "required" : true
+          "description" : "Script language"
```

Comment:
Its not quite the `Script language` as `string`.  It's a new paramater that will cause the passed script to be compiled and evaluated in a certain `ScriptContext`. 

See: https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/script/ScriptModule.java#L41-L52

This should be an `enum` with multiple values:

`search`, `aggs`, `executable`, `aggs_executable`, `update`, `ingest`, `filter`, `similarity`, `similarity_weight` & `template`. 

Thats the current list for `master` `6.0` and `6.x`  have a slightly smaller list:

https://github.com/elastic/elasticsearch/blob/6.0/core/src/main/java/org/elasticsearch/script/ScriptModule.java#L41-L52

https://github.com/elastic/elasticsearch/blob/6.x/core/src/main/java/org/elasticsearch/script/ScriptModule.java#L41-L52

Reply:
An enum is not correct. These values are pluggable.

Reply:
@Mpdreamz thanks for putting it in context!

Reply:
Thanks @rjernst, @olcbean LGTM now 👍